pkg/assembler/backends/backends.go

```diff
@@ -71,6 +71,10 @@ type Backend interface {
 	IngestHasSourceAt(ctx context.Context, pkg model.PkgInputSpec, pkgMatchType model.MatchFlags, source model.SourceInputSpec, hasSourceAt model.HasSourceAtInputSpec) (*model.HasSourceAt, error)
 	IngestIsVulnerability(ctx context.Context, osv model.OSVInputSpec, vulnerability model.CveOrGhsaInput, isVulnerability model.IsVulnerabilityInputSpec) (*model.IsVulnerability, error)
 	IngestVEXStatement(ctx context.Context, subject model.PackageOrArtifactInput, vulnerability model.CveOrGhsaInput, vexStatement model.VexStatementInputSpec) (*model.CertifyVEXStatement, error)
+
+	// More advanced queries
+	Path(ctx context.Context, subject string, target string, maxPathLength int) ([]model.Node, error)
+	Neighbors(ctx context.Context, node string) ([]model.Node, error)
```

Comment:
nit should this be uint32?

Reply:
It should, but there's a bug in gqlgen that now makes all IDs convert to string scalars. Waiting on a fix there and the next codegen will ask us to resolve